test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java

```diff
@@ -1206,4 +1214,67 @@ private void testFailoverWithStreamSubsumedByCheckpoint(boolean trim) throws Exc
         shutdownCorfuServer(server1);
         shutdownCorfuServer(server2);
     }
+
+    @Test
+    public void nodeResetPruneRaceTest() throws Exception {
+        final int PORT_0 = 9000;
+        final int PORT_1 = 9001;
+        final int PORT_2 = 9002;
+        final Duration timeout = Duration.ofMinutes(5);
+        final Duration pollPeriod = Duration.ofSeconds(5);
+        final int workflowNumRetry = 3;
+
+        Process corfuServer_1 = runPersistentServer(corfuSingleNodeHost, PORT_0, true);
+        Process corfuServer_2 = runPersistentServer(corfuSingleNodeHost, PORT_1, false);
+        Process corfuServer_3 = runPersistentServer(corfuSingleNodeHost, PORT_2, false);
+
+        runtime = createDefaultRuntime();
+
+        CorfuTable<String, String> table = runtime.getObjectsView()
+                .build()
+                .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+                .setStreamName("test")
+                .open();
+
+
+        //fetch the layout without the client for localhost:9001
+        runtime.invalidateLayout();
+
+        final AtomicBoolean moreDataToBeWritten = new AtomicBoolean(true);
+
+        boolean result = true;
+        int retry = 0;
+
+        //Without the fix, this will retry 3 times.
+        while (result && retry < NUM_RETRY) {
+            result = false;
+            try {
+                log.debug("\n*****retry : " + retry + " clientMap " + runtime.getLayoutView().getRuntimeLayout().getSenderClientMap());
+                BaseClient baseClient = runtime.getLayoutView().getRuntimeLayout()
+                        .getBaseClient("localhost:9001");
+                IClientRouter router = baseClient.getRouter();
+                log.debug("\nrouter " + router + " map " + runtime.getLayoutView().getRuntimeLayout().getSenderClientMap());
+                if (retry == 0) {
+                    runtime.invalidateLayout();
+                }
+                router.sendMessageAndGetCompletable(new CorfuMsg(CorfuMsgType.RESET)
```

Comment:
I think it would be nice to simulate the layout invalidation concurrently somehow while resetting a node. I know that this is very hard to make it in a such a way that it produces the same test output all the time especially with our code structure but I think it will move us closer to the reproduction.